lib/file_transfer_agent/gcs_util.js

```diff
@@ -445,26 +442,39 @@ function GCSUtil(httpclient, filestream) {
   /**
   * Generate file URL based on bucket.
   *
-  * @param {String} stageLocation
+  * @param {Object} stageInfo
   * @param {String} filename
   *
   * @returns {String}
   */
-  this.generateFileURL = function (stageLocation, filename) {
-    const gcsLocation = this.extractBucketNameAndPath(stageLocation);
+  this.generateFileURL = function (stageInfo, filename) {
+    const gcsLocation = this.extractBucketNameAndPath(stageInfo.location);
     const fullFilePath = `${gcsLocation.path}${filename}`;
-    const link = 'https://storage.googleapis.com/' + gcsLocation.bucketName + '/' + fullFilePath;
+    const endPoint = this.getGCSCustomEndPoint(stageInfo);
+    const link = `${endPoint != null ? endPoint : 'https://storage.googleapis.com'}/${gcsLocation.bucketName}/${fullFilePath}`;
     return link;
   };
 
+  this.getGCSCustomEndPoint = function (stageInfo) {
+    //TODO: SNOW-1789759 hardcoded region will be replaced in the future
+    const isRegionalUrlEnabled = (stageInfo.region).toLowerCase() === 'me-central2' || stageInfo.useRegionalUrl;
+    let endPoint = null;
+    if (stageInfo['endPoint']) {
+      endPoint = stageInfo['endPoint'];
+    } else if (isRegionalUrlEnabled) {
+      endPoint = `storage.${stageInfo.region.toLowerCase()}.rep.googleapis.com`;
+    }
+    return endPoint;
+  };
+
   /**
-* Left strip the specified character from a string.
-*
-* @param {String} str
-* @param {Character} remove
-*
-* @returns {String}
-*/
+  * Left strip the specified character from a string.
+  *
+  * @param {String} str
+  * @param {Character} remove
+  *
+  * @returns {String}
+  */
   function lstrip(str, remove) {
```

Comment:
Can we add some simple unit test for this method?

Reply:
I moved this function to Util.js and created testing.